controllers/windowsmachine_controller.go

```diff
@@ -106,9 +106,9 @@ func (r *WindowsMachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		UpdateFunc: func(e event.UpdateEvent) bool {
 			return r.isValidMachine(e.ObjectNew) && isWindowsMachine(e.ObjectNew.GetLabels())
 		},
-		// ignore delete event for all Machines as WMCO does not react to node getting deleted
+		// process delete event
 		DeleteFunc: func(e event.DeleteEvent) bool {
-			return false
+			return r.isValidMachine(e.Object) && isWindowsMachine(e.Object.GetLabels())
```

Comment:
After looking closely, is okay and safer to include `r.isValidMachine()` while filtering with delete predicate. Machines in `Deleting` phase does maintain all fields populated, as @mansikulkarni96  [pointed out](https://github.com/openshift/windows-machine-config-operator/pull/571#discussion_r689601348).


```
oc describe machine winworker-vfbzb -n openshift-machine-api
Name:         winworker-vfbzb
Namespace:    openshift-machine-api
Labels:       machine.openshift.io/cluster-api-cluster=jvaldes-wl2pk
              machine.openshift.io/cluster-api-machine-role=worker
              machine.openshift.io/cluster-api-machine-type=worker
              machine.openshift.io/cluster-api-machineset=winworker
              machine.openshift.io/os-id=Windows
              machine.openshift.io/region=
              machine.openshift.io/zone=
Annotations:  machine.openshift.io/instance-state: poweredOn
API Version:  machine.openshift.io/v1beta1
Kind:         Machine
Metadata:
  Creation Timestamp:             2021-08-16T20:36:50Z
  Deletion Grace Period Seconds:  0
  Deletion Timestamp:             2021-08-16T20:40:32Z
  Finalizers:
    machine.machine.openshift.io
  Generate Name:  winworker-
  Generation:     3
  Managed Fields:{}
  Owner References:
    API Version:           machine.openshift.io/v1beta1
    Block Owner Deletion:  true
    Controller:            true
    Kind:                  MachineSet
    Name:                  winworker
    UID:                   2ab5c21a-c6ce-4000-b777-0f04dfa7dc7a
  Resource Version:        141193
  UID:                     01f725ed-24cc-433b-8654-f8077dc9556f
Spec:
  Metadata:
    Labels:
      node-role.kubernetes.io/worker:  
  Provider ID:                         vsphere://422cbd84-025f-0a2a-dab1-cac6bce7d16f
  Provider Spec:
    Value:
      API Version:  vsphereprovider.openshift.io/v1beta1
      Credentials Secret:
        Name:       vsphere-cloud-credentials
      Disk Gi B:    120
      Kind:         VSphereMachineProviderSpec
      Memory Mi B:  8192
      Metadata:
        Creation Timestamp:  <nil>
      Network:
        Devices:
          Network Name:      dev-segment
      Num CP Us:             4
      Num Cores Per Socket:  1
      Snapshot:              
      Template:              windows-golden-images/windows-server-2004-template
      User Data Secret:
        Name:  worker-user-data
      Workspace:
        Datacenter:     SDDC-Datacenter
        Datastore:      WorkloadDatastore
        Folder:         /SDDC-Datacenter/vm/jvaldes-wl2pk
        Resource Pool:  /SDDC-Datacenter/host/Cluster-1/Resources
        Server:         vcenter.sddc-44-236-21-251.vmwarevmc.com
Status:
  Addresses:
    Address:  fe80::cd5b:e920:b625:a5bd
    Type:     InternalIP
    Address:  172.31.251.49
    Type:     InternalIP
    Address:  winworker-vfbzb
    Type:     InternalDNS
  Conditions:
    Last Transition Time:  2021-08-16T20:37:24Z
    Status:                True
    Type:                  InstanceExists
  Last Updated:            2021-08-16T20:40:33Z
  Phase:                   Deleting
  Provider Status:
    Conditions:
      Last Probe Time:       2021-08-16T20:36:53Z
      Last Transition Time:  2021-08-16T20:36:53Z
      Message:               Machine successfully created
      Reason:                MachineCreationSucceeded
      Status:                True
      Type:                  MachineCreation
    Instance Id:             422cbd84-025f-0a2a-dab1-cac6bce7d16f
    Instance State:          poweredOn
    Task Ref:                task-3905531
Events:
  Type     Reason               Age                    From               Message
  ----     ------               ----                   ----               -------
  Normal   Create               3m47s                  vspherecontroller  Created Machine winworker-vfbzb
  Warning  FailedUpdate         3m25s (x2 over 3m43s)  vspherecontroller  winworker-vfbzb: reconciler failed to Update machine: task task-3905478 has not finished
  Warning  MachineSetupFailure  98s                    windowsmachine     Machine winworker-vfbzb configuration failure
  Normal   Update               7s (x9 over 3m16s)     vspherecontroller  Updated Machine winworker-vfbzb
  Warning  FailedDelete         7s                     vspherecontroller  winworker-vfbzb: reconciler failed to Delete machine: destroying vm in progress, reconciling
  Warning  FailedDelete         6s (x2 over 7s)        vspherecontroller  winworker-vfbzb: reconciler failed to Delete machine: task task-3905531 has not finished
  Normal   Delete               0s                     vspherecontroller  Deleted machine winworker-vfbzb
```